extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java

```diff
@@ -654,7 +654,9 @@ private List<IClasspathEntry> getDependencyJars(DependencyModulesResult dependen
      */
     private String getHighestCompatibleJavaVersion(String gradleVersion) {
       GradleVersion version = GradleVersion.version(gradleVersion);
-      if (version.compareTo(GradleVersion.version("8.3")) >= 0) {
+      if (version.compareTo(GradleVersion.version("8.5")) >= 0) {
+        return "21"; // JavaCore.VERSION_21 not available yet
```

Comment:
`VERSION_21` should be available. Assume you are using VS Code for development, you can try:

1. install the [PDE support extension](https://marketplace.visualstudio.com/items?itemName=yaozheng.vscode-pde)
2. Open the project
3. The PDE extension should be able to load the target platform definition automatically from the `javaConfig.json`
4. If it's still not available. Reload the window, and click `Reload target platform` as below:
![image](https://github.com/microsoft/vscode-gradle/assets/6193897/67554e15-0f8c-4375-bf93-b642e5cab403)

Sorry looks like this is not mentioned in the contribution.md

Reply:
Excellent, thanks for the info! I have changed the literal to `JavaCore.VERSION_21` 🙂